rls/src/test/java/io/grpc/rls/internal/RlsLoadBalancerTest.java

```diff
@@ -0,0 +1,578 @@
+import io.grpc.LoadBalancer.CreateSubchannelArgs;
+import io.grpc.LoadBalancer.Helper;
+import io.grpc.LoadBalancer.PickResult;
+import io.grpc.LoadBalancer.ResolvedAddresses;
+import io.grpc.LoadBalancer.Subchannel;
+import io.grpc.LoadBalancer.SubchannelPicker;
+import io.grpc.LoadBalancer.SubchannelStateListener;
+import io.grpc.ManagedChannel;
+import io.grpc.Metadata;
+import io.grpc.MethodDescriptor;
+import io.grpc.MethodDescriptor.Marshaller;
+import io.grpc.MethodDescriptor.MethodType;
+import io.grpc.NameResolver.ConfigOrError;
+import io.grpc.NameResolver.Factory;
+import io.grpc.Status;
+import io.grpc.SynchronizationContext;
+import io.grpc.inprocess.InProcessChannelBuilder;
+import io.grpc.inprocess.InProcessServerBuilder;
+import io.grpc.internal.JsonParser;
+import io.grpc.internal.PickSubchannelArgsImpl;
+import io.grpc.lookup.v1.RouteLookupServiceGrpc;
+import io.grpc.rls.RlsLoadBalancerProvider;
+import io.grpc.rls.internal.CachingRlsLbClient.RlsPicker;
+import io.grpc.rls.internal.RlsProtoConverters.RouteLookupResponseConverter;
+import io.grpc.rls.internal.RlsProtoData.RouteLookupRequest;
+import io.grpc.rls.internal.RlsProtoData.RouteLookupResponse;
+import io.grpc.stub.StreamObserver;
+import io.grpc.testing.GrpcCleanupRule;
+import java.io.IOException;
+import java.lang.Thread.UncaughtExceptionHandler;
+import java.net.SocketAddress;
+import java.util.Collections;
+import java.util.Deque;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import javax.annotation.Nonnull;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.AdditionalAnswers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.InOrder;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.junit.MockitoJUnit;
+import org.mockito.junit.MockitoRule;
+
+@RunWith(JUnit4.class)
+public class RlsLoadBalancerTest {
+
+  @Rule
+  public final GrpcCleanupRule grpcCleanupRule = new GrpcCleanupRule();
+  @Rule
+  public final MockitoRule mocks = MockitoJUnit.rule();
+
+  private final DoNotUseDirectScheduledExecutorService fakeScheduledExecutorService =
+      mock(DoNotUseDirectScheduledExecutorService.class, CALLS_REAL_METHODS);
+  private final SynchronizationContext syncContext =
+      new SynchronizationContext(new UncaughtExceptionHandler() {
+        @Override
+        public void uncaughtException(Thread t, Throwable e) {
+          throw new RuntimeException(e);
+        }
+      });
+  private final Helper helper =
+      mock(Helper.class, AdditionalAnswers.delegatesTo(new FakeHelper()));
+  private final FakeRlsServerImpl fakeRlsServerImpl = new FakeRlsServerImpl();
+  private final Deque<FakeSubchannel> subchannels = new LinkedList<>();
```

Comment:
Probably will see lint warnings for using LinkedList rather than ArrayDequeue when import.

Reply:
yeah, it is pretty bad they added that check. we should generally ignore it unless it is indeed right, because the warning is you are **maybe** doing wrong.